clientv3/example_kv_test.go

```diff
@@ -259,6 +259,43 @@ func ExampleKV_txn() {
 	// Output: key : XYZ
 }
 
+func ExampleKV_txn_notexist() {
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   endpoints,
+		DialTimeout: dialTimeout,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer cli.Close()
+
+	kvc := clientv3.NewKV(cli)
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	_, err = kvc.Txn(ctx).
+	// txn value comparisons are lexical
+		If(clientv3.CompareExist(clientv3.Value("key"), "ne")).
```

Comment:
Sorry I am still confused. If we replace this line with the following line, isn't it the same?
`If(clientv3.Compare(clientv3.ModRevision("key"), "=", 0))`

Reply:
BTW, I don't think this PR solves the original issue #10566. The original issue is that when we use clientv3.Compare(clientv3.Value(key), "=", value), the compare always return `false` if key does not exist, regardless of "=" or "!=".

Reply:
Thank you for your suggestions.

1: There are the same, i just think `ModRevision` make user realize this API should be use as Optimistic lock in mvcc storage


2: `clientv3.Compare(clientv3.Value(key), "=", value)` always return false, the reason just is 

https://github.com/etcd-io/etcd/blob/dc6885d73f71d4f7e77007fcc794f36e09fa5ecb/etcdserver/apply.go#L449

This PR had delete these codes.
I am also confused why add these codes, and think value compare result should be decided by bytes.Compare

Reply:
> Thank you for your suggestions.
> 
> 1: There are the same, i just think `ModRevision` make user realize this API should be use as Optimistic lock in mvcc storage

Thanks for confirming. I think we should not break API to implement a feature that already exists.

> 2: `clientv3.Compare(clientv3.Value(key), "=", value)` always return false, the reason just is
> 
> https://github.com/etcd-io/etcd/blob/dc6885d73f71d4f7e77007fcc794f36e09fa5ecb/etcdserver/apply.go#L449
> 
> This PR had delete these codes.
> I am also confused why add these codes, and think value compare result should be decided by bytes.Compare

Sorry I missed the deleting part. Removing the code block means treating non-exist-key's value as an empty string (or its byte presentation). This is probably fine, but we should more carefully think through this.